### PR TITLE
Allow multiple links with custom link field

### PR DESCRIPTION
**Pull request by ehelms**
I realize this is a one-off patch that I am more than happy to try and contribute to upstream Redmine. However, given what it will give us with respect to issue and PR management I'd like to get it in for our use sooner than the cycle it may take to get it into upstream Redmine. I didn't test with a replica DB, but I did use this development container (https://github.com/Katello/katello-deploy/pull/185) to mount Redmine, add the changes here, add a custom Link field with multiple values set to true and create some issues with multiple links through the UI and modify them via the API.

### CODE REVIEW

File: lib/redmine/field_format.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def cast_single_value(custom_field, value, customized=nil)
       def target_class
         nil
       end
- 
+
```
Comment (domcleal) [inline lib/redmine/field_format.rb]: Could you keep this, just to reduce the change of merge conflicts when pulling from upstream?

File: lib/redmine/field_format.rb [hunk 2]
```diff
@@ -496,7 +518,7 @@ def check_box_edit_tag(view, tag_id, tag_name, custom_value, options={})
           tag = view.send(tag_method, tag_name, value, checked, :id => tag_id)
           # set the id on the first tag only
           tag_id = nil
-          s << view.content_tag('label', tag + ' ' + label)
```
Comment (domcleal) [inline lib/redmine/field_format.rb]: ditto on these others

**Comment by domcleal** [comment]
Works very well indeed.  Please do open a ticket on Redmine's own tracker and attach the patch, hopefully they will be interested, but it can be hit and miss I think.  It should be quite easy to start the process even if it doesn't go anywhere.  I see it isn't in master right now.

I'll need a corresponding update to the prprocessor in order to merge this as the format of the API response at least has changed.

**Comment by ehelms** [comment]
Here is the corresponding prprocessor PR -- https://github.com/theforeman/prprocessor/pull/30
Associated Redmine upstream issue and patch -- http://www.redmine.org/issues/21115

**Comment by domcleal** [comment]
Thanks @ehelms, merging now.

**Comment by domcleal** [comment]
You'll need to add f628046 into your Redmine patch, as old tickets that predate the addition of the custom field have nil values.  The string part of the conditional can understand it, but the array one failed with http://fpaste.org/285204/44621290/.

**Pull request merged**